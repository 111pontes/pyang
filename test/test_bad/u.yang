module u {
  yang-version 1.1;
  namespace urn:u;
  prefix u;

  feature foo;
  feature bar;

  container a {
    if-feature "foo and not bar or foox"; // error: foox not defined
  }

  container b {
    if-feature "foo and not"; // error: bad syntax
  }

  container c {
    if-feature "foo and "; // error: bad syntax
  }

  action reset; // error: not allowed on top-level

  typedef id {
    type string {
      pattern '[a-zA-Z_][a-zA-Z0-9\-_.]*';
      pattern '[xX][mM][lL].*' {
        modifier invert-match;
      }
    }
  }

  leaf foo {
    type id;
    default "xMlu"; // error: bad pattern
  }

  leaf bar {
    type id;
    default "xmq-foo";
  }
}
