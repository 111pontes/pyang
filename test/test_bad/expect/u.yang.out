u.yang:10: error: FEATURE_NOT_FOUND
u.yang:14: error: BAD_VALUE
u.yang:18: error: BAD_VALUE
u.yang:21: error: UNEXPECTED_KEYWORD
u.yang:34: error: TYPE_VALUE
u.yang:63: error: IDENTITY_NOT_FOUND
